frontend/packages/kubevirt-plugin/src/components/vms/vm-environment/selectors.ts

```diff
@@ -29,6 +32,33 @@ export const getSourceName = (ed: EnvDisk): string => {
   );
 };
 
+const getVolumeBySource = (sourceName: string, vmWrapper: VMWrapper) => {
+  return vmWrapper.getVolumes().find((vol) => {
+    const volWrapper = new VolumeWrapper(vol);
+    const volType = volWrapper.getType();
+
+    if (!volType.isEnvType()) {
+      return false;
+    }
+
+    switch (volType) {
```

Comment:
this is the exact logic of `getVolumeReferencedObject`. So you can do `volWrapper.getVolumeReferencedObject()?.name === sourceName` instead

and isEnvType is ensuring you are having the correct type